content/en/blog/_posts/2023-03-10-image-registry-change.md

```diff
@@ -0,0 +1,190 @@
+```
+
+## How can I find which images are using the legacy registry?
+
+**Option 1**: See the one line kubectl command in our [earlier blog
+post](https://kubernetes.io/blog/2023/02/06/k8s-gcr-io-freeze-announcement/#what-s-next):
+
+```
+kubectl get pods --all-namespaces -o jsonpath="{.items[*].spec.containers[*].image}" |\
+tr -s '[[:space:]]' '\n' |\
+sort |\
+uniq -c
+```
+
+**Option 2**: A `kubectl` [krew](https://krew.sigs.k8s.io/) plugin has been developed called
+[`community-images`](https://github.com/kubernetes-sigs/community-images#kubectl-community-images),
+that will scan and report any images using the k8s.gcr.io endpoint.
+
+If you have krew installed, you can install it with:
+
+```
+kubectl krew install community-images
+```
+
+and generate a report with:
+
+```
+kubectl community-images
+```
+
+For alternate methods of install and example output, check out the repo:
+[kubernetes-sigs/community-images](https://github.com/kubernetes-sigs/community-image).
+
+**Option 3**: If you do not have access to a cluster directly, or manage many clusters - the best
+way is to run a search over your manifests and charts for _"k8s.gcr.io"_.
+
+**Option 4**: If you wish to prevent k8s.gcr.io based images from running in your cluster, example
+policies for [Gatekeeper](https://open-policy-agent.github.io/gatekeeper-library/website/) and
+[Kyverno](https://kyverno.io/) are available in the [AWS EKS Best Practices
+repository](https://github.com/aws/aws-eks-best-practices/tree/master/policies/k8s-registry-deprecation)
+that will block them from being pulled. You can use these third-party policies with any Kubernetes
+cluster.
+
+**Option 5**: As a **LAST** possible option, you can use a [Mutating Admission
+Webhook](https://github.com/slackhq/simple-kubernetes-webhook/blob/main/dev/gen-certs.sh) to change
+the image address dynamically. This should only be considered a stopgap till your manifests have
+been updated. An example Mutating Webhook and Kyverno policy can be found
+[here](https://github.com/abstractinfrastructure/k8s-gcr-quickfix).
+
+## I still have questions, where should I go?
+
+For more information on registry.k8s.io and why it was developed, see [registry.k8s.io: faster,
+cheaper and Generally Available
+(GA)](https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/).
+
+If you would like to know more about the image freeze and the last images that will be available
+there, see the blog post: [k8s.gcr.io Image Registry Will Be Frozen From the 3rd of April
+2023](https://kubernetes.io/blog/2023/02/06/k8s-gcr-io-freeze-announcement/).
+
+Information on the architecture of registry.k8s.io and its [request handling decision
+tree](https://github.com/kubernetes/registry.k8s.io/blob/main/cmd/archeio/docs/request-handling.md)
```

Comment:
```suggestion
tree](https://github.com/kubernetes/registry.k8s.io/blob/8408d0501a88b3d2531ff54b14eeb0e3c900a4f3/cmd/archeio/docs/request-handling.md)
```

Permalinking in case that file ever moves